pareto/utilities/cm_utils/run_utils.py

```diff
@@ -0,0 +1,432 @@
+    # opt.options['NonConvex'] = 2
+    # opt.options['TimeLimit'] = 150
+    status = opt.solve(model, tee=False)
+    term_cond = pyo.check_optimal_termination(status)
+    if not term_cond:
+        print(
+            "Linear model is infeasibility. This likely means the production flow \
+              exceeds the disposal and treatment capacities"
+        )
+
+    # unfixing all the initialized variables
+    # model = obj_fix(model, [], deactivate=[], activate= conclist+flowlist)
+    model = alter_nonlinear_cons(model, deactivate=False)
+
+    # running bilinear model
+    print("\nSolving bilinear problem...")
+    opt = pyo.SolverFactory("ipopt")
+
+    opt.options["max_iter"] = max_iterations
+    status = opt.solve(model, tee=tee)
+    model.status = status
+
+    # ------------------------------Results------------------------------------
+
+    # showing results
+    term_cond = pyo.check_optimal_termination(status)
+    if term_cond == True:
+
+        print("\n Successfully solved model")
+        # Displaying specific broken down costs
+        print_results_summary(model)
+
+        # returning specific broken down costs
+        value_strings = [
+            "Arc cost",
+            "Disposal cost",
+            "Freshwater cost",
+            "Treatment cost",
+            "Storage cost",
+            "Storage revenue",
+            "Treatment revenue",
+            "\nTotal cost",
+        ]
+
+        values = [
+            model.arc_cost,
+            model.disp_cost,
+            model.fresh_cost,
+            model.treat_cost,
+            model.stor_cost,
+            model.stor_rev,
+            model.treat_rev,
+            model.obj() * 1000,
+        ]
+        values = [pyo.value(x) for x in values]
+        values = {name: value for name, value in zip(value_strings, values)}
+        return model, values
+
+    # returning values of 0s if model was not feasible
+    else:
+        print("Model is Infeasible")
+        value_strings = [
+            "Arc cost",
+            "Disposal cost",
+            "Freshwater cost",
+            "Treatment cost",
+            "Storage cost",
+            "Storage revenue",
+            "Treatment revenue",
+            "\nTotal cost",
+        ]
+
+        values = [0, 0, 0, 0, 0, 0, 0, 0 * 1000]
+
+        if inf_recs:
+            print(
+                "\
+                The model is infeasible. Some possible reasons for infeasibility: \n \
+                1. The minimum CM recovery concentration is too high given the existing infrastructure and parameters \n \
+                2. The produced water concentration is too low to meet minimum CM concentration requirements \n \
+                3. The treatment efficiency is not sufficient to meet the minimum CM concentration \n \
+                4. Installation of pipelines for direct transportation of water from high CM concentration production pads to desalination sites would aid feasibility \n \
```

Comment:
```suggestion
                Suggestion to improve feasibility: Allow installation of pipelines for direct transportation of water from high CM concentration production pads to desalination sites \n \
```
This last item isn't a possible reason for infeasibility, but a suggestion about what to do.